dali/pipeline/operator/direct_operator.h

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATOR_DIRECT_OPERATOR_H_
+#define DALI_PIPELINE_OPERATOR_DIRECT_OPERATOR_H_
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include "dali/core/cuda_stream_pool.h"
+#include "dali/pipeline/data/tensor_list.h"
+#include "dali/pipeline/operator/op_spec.h"
+#include "dali/pipeline/operator/operator.h"
+#include "dali/pipeline/util/backend2workspace_map.h"
+#include "dali/pipeline/util/thread_pool.h"
+#include "dali/pipeline/workspace/workspace.h"
+
+namespace dali {
+
+template <typename Backend>
+std::shared_ptr<TensorList<Backend>> AsTensorList(std::shared_ptr<TensorList<Backend>> input) {
+  return input;
+}
+
+template <typename Backend>
+std::shared_ptr<TensorList<Backend>> AsTensorList(std::shared_ptr<TensorVector<Backend>> input) {
+  // TODO(ksztenderski): Remove copy.
+  auto tl = std::make_shared<TensorList<Backend>>();
+  tl->Copy(*input);
+  return tl;
+}
```

Comment:
We have a very similar thing in `workspace_policy.h` - `PresentAsTensorList`. Maybe it should be unified and moved to a common untility.

Reply:
We are doing a copy here (I guess for a prototype) which will always work but the `PresentAsTensorList` requires contiguous TV to work.

Reply:
It can be another flavor or option of that function.

Reply:
done